batch_object_tagger.tf

```
@@ -0,0 +1,100 @@
+locals {
+  pdm_object_tagger_image = "${local.account.management}.${data.terraform_remote_state.aws_ingestion.outputs.vpc.vpc.ecr_dkr_domain_name}/dataworks-s3-object-tagger:${var.image_version.s3-object-tagger}"
+}
+
+# AWS Batch Job IAM role
+data "aws_iam_policy_document" "batch_assume_policy" {
+  statement {
+    sid    = "BatchAssumeRolePolicy"
+    effect = "Allow"
+
+    actions = [
+      "sts:AssumeRole",
+    ]
+
+    principals {
+      identifiers = ["ecs-tasks.amazonaws.com"]
+
+      type = "Service"
+    }
+  }
+}
+
+resource "aws_iam_role" "pdm_object_tagger" {
+  name               = "pdm_object_tagger"
+  assume_role_policy = data.aws_iam_policy_document.batch_assume_policy.json
+  tags               = local.common_tags
+}
+
+data "aws_iam_policy_document" "pdm_object_tagger" {
+  statement {
+    sid    = "AllowS3Tagging"
+    effect = "Allow"
+
+    actions = [
+      "s3:ListObjects",
+      "s3:GetObject",
+      "s3:*Tagging"
+    ]
+
+    resources = [
+      data.terraform_remote_state.adg.outputs.published_bucket.arn,
```

object permissions apply to the bucket contents, rather than the bucket
```suggestion
      "${data.terraform_remote_state.adg.outputs.published_bucket.arn}/*",
```